crates/catalog/rest/src/catalog.rs

```diff
@@ -0,0 +1,845 @@
+            .build()?;
+
+        self.client
+            .execute::<ErrorModel, NO_CONTENT>(request)
+            .await
+            .map(|_| true)
+    }
+
+    /// Rename a table in the catalog.
+    async fn rename_table(&self, src: &TableIdent, dest: &TableIdent) -> Result<()> {
+        let request = self
+            .client
+            .0
+            .post(self.config.rename_table_endpoint())
+            .json(&RenameTableRequest {
+                source: src.clone(),
+                destination: dest.clone(),
+            })
+            .build()?;
+
+        self.client.execute::<ErrorModel, NO_CONTENT>(request).await
+    }
+
+    /// Update a table to the catalog.
+    async fn update_table(&self, _table: &TableIdent, _commit: TableCommit) -> Result<Table> {
+        todo!()
+    }
+
+    /// Update multiple tables to the catalog as an atomic operation.
+    async fn update_tables(&self, _tables: &[(TableIdent, TableCommit)]) -> Result<()> {
+        todo!()
+    }
+}
+
+impl RestCatalog {
+    /// Creates a rest catalog from config.
+    pub async fn new(config: RestCatalogConfig) -> Result<Self> {
+        let mut catalog = Self {
+            client: config.try_create_rest_client()?,
+            config,
+        };
+
+        catalog.update_config().await?;
+        catalog.client = catalog.config.try_create_rest_client()?;
+
+        Ok(catalog)
+    }
+
+    async fn update_config(&mut self) -> Result<()> {
+        let mut request = self.client.0.get(self.config.config_endpoint());
+
+        if let Some(warehouse_location) = &self.config.warehouse {
+            request = request.query(&[("warehouse", warehouse_location)]);
+        }
+        let mut config = self
+            .client
+            .query::<CatalogConfig, ErrorResponse, OK>(request.build()?)
+            .await?;
+
+        config.defaults.extend(self.config.props.clone());
+        config.defaults.extend(config.overrides);
+
+        self.config.props = config.defaults;
+
+        Ok(())
+    }
+}
+
+/// Requests and responses for rest api.
+mod _serde {
+    use std::collections::HashMap;
+
+    use serde_derive::{Deserialize, Serialize};
+
+    use iceberg::{Error, ErrorKind, Namespace, TableIdent};
+
+    pub(super) const OK: u16 = 200u16;
+    pub(super) const NO_CONTENT: u16 = 204u16;
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub(super) struct CatalogConfig {
+        pub(super) overrides: HashMap<String, String>,
+        pub(super) defaults: HashMap<String, String>,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    pub(super) struct ErrorResponse {
+        error: ErrorModel,
+    }
+
+    impl From<ErrorResponse> for Error {
+        fn from(resp: ErrorResponse) -> Error {
+            resp.error.into()
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    pub(super) struct ErrorModel {
+        pub(super) message: String,
+        pub(super) r#type: String,
+        pub(super) code: i32,
```

Comment:
Nit: We use u16 for the error code in other places.
```suggestion
        pub(super) code: u16,
```